app/decorators/sufficient_quota_decorator.rb

```diff
@@ -0,0 +1,43 @@
+# Extend functionality of {Quota} object to describe if there are sufficient
+# disk resources still available for the given volume
+class SufficientQuotaDecorator < SimpleDelegator
+  # Threshold percentage that defines whether there are still sufficient disk
+  # resources remaining
+  # @return [Float] threshold percentage
+  attr_reader :threshold
+
+  # @param quota [Quota] the quota object we want to decorate
+  # @param opts [#to_h] initialization options
+  # @option opts [#to_f] :threshold threshold percentage
+  def initialize(quota, opts = {})
+    super(quota)
+
+    opts = opts.to_h.compact.symbolize_keys
+    @threshold = opts.fetch(:threshold, Configuration.quota_threshold).to_f
```

Comment:
Configuration.quota_threshold should not be in this class. We only instantiate SufficientQuotaDecorator in one place, so I would do this:

```
SufficientQuotaDecorator.new(quota, threshold: Configuration.quota_threshold)
```

I'm certain I have done this same thing in other areas of the code and I should fix those.